docs/service_notifications.md

```diff
@@ -0,0 +1,199 @@
+lower than the one received in this event.
+
+For other services,
+there is no way to know the generation number
+associated with a previous login.
+It's probably best to ignore `generation`
+and instead terminate login sessions
+established prior to the timestamp in `ts`.
+
+
+### Password Change Event
+
+Message Properties:
+
+* `event`: The string "passwordChange".
+* `uid`: The userid of the account that had its password changed.
+* `iss`: The API server that issued the event ("api.accounts.firefox.com" in production environments)
+* `generation`: A monotonically increasing integer for de-duping events (ok ok, it's a timestamp).
+
+As with the "reset" event above,
+services receiving this event should
+terminate user login sessions
+that were established prior to the event.
+
+
+### Change of Primary Email Address Event
+
+Message Properties:
+
+* `event`: The string "primaryEmailChanged".
+* `uid`: The userid of the account that was updated.
+* `email`: The email address that is the new primary for the account.
+
+Note that the message does *not* include
+the previous primary email address.
+Receiving services should use the `uid`
+to uniquely identify a user,
+and should update their records
+of that user's preferred email address.
+
+
+### Change of Profile Data
+
+Message Properties:
+
+* `event`: The string "profileDataChange".
+* `uid`: The userid of the account that was updated.
+
+Receiving services should update any cached profile information
+that they hold about the user,
+such as their display name or 2FA status.
+
+
+### Device Connection Event
+
+Message Properties:
+
+* `event`: The string "device:create".
+* `uid`: The userid of the account to which a device was connected.
+* `id`: The id of the device that was connected.
+* `timestamp`: Integer millisecond timestamp at which the event occurred.
+* `type`: The type of device, e.g. "desktop" or "mobile".
+* `isPlaceholder`: Boolean indicating whether the device explicitly registered itself,
+   or had a device record implicitly created by the server.
+
+Services that know about connected devices should,
+upon receiving this event,
+make accommodations for any state
+that may need to be stored for that device
+
+Sync is the only service
+that currently knows about connected devices,
+so other services probably shouldn't use this.
+
+
+### Device Disconnection Event
+
+Message Properties:
+
+* `event`: The string "device:delete".
+* `uid`: The userid of the account from which a device was disconnected.
+* `id`: The id of the device that was disconnected.
+* `timestamp`: Integer millisecond timestamp at which the event occurred.
+
+Services that know about connected devices should,
+upon receiving this event,
+discard any transient state stored for that device
+and deny any future access requests
+made on behalf of that device.
+
+Sync is the only service
+that currently knows about connected devices,
+so other services probably shouldn't use this.
```

Comment:
Can we link to examples of where we consume these events? I think we have a node.js and a Python example in the wild.